hashes/src/hmac.rs

```diff
@@ -162,6 +162,17 @@ impl<'de, T: GeneralHash + Deserialize<'de>> Deserialize<'de> for Hmac<T> {
     }
 }
 
+#[cfg(any(feature = "std", feature = "bitcoin-io"))]
```

Comment:
It'd be simpler to just let the macro produce nothing if the features are off.

Reply:
I did this to make it clear to the casual reader that `impl_io_write` is only called when either `std` or `bitcoin-io` are enabled instead of them seeing the macro called and thinking io stuff is unconditionally implemented.

Reply:
It breaks SSOT - you now have to both handle the edge case when calling the macro and call the macro even though the macro itself could've handled it all.